From 668ca11c3064310077c5354c45d54faf81ee2385 Mon Sep 17 00:00:00 2001
From: Tom Rix <trix@redhat.com>
Date: Sat, 23 Dec 2023 08:27:15 -0500
Subject: [PATCH] Prepare torchtext cmake for Fedora

remove the third_party dir
version the *.so
set build type to RelWithDebInfo

Signed-off-by: Tom Rix <trix@redhat.com>
---
 CMakeLists.txt                |  3 ++-
 torchtext/csrc/CMakeLists.txt | 10 +---------
 2 files changed, 3 insertions(+), 10 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index b789a3226017..1669e4874a59 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -18,6 +18,8 @@ endif()
 
 project(torchtext)
 
+set(CMAKE_BUILD_TYPE "RelWithDebInfo" CACHE STRING "Need some debug info" FORCE)
+
 
 # check and set CMAKE_CXX_STANDARD
 string(FIND "${CMAKE_CXX_FLAGS}" "-std=c++" env_cxx_standard)
@@ -63,5 +65,4 @@ endif()
 
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${TORCH_COMPILED_WITH_CXX_ABI} -Wall ${TORCH_CXX_FLAGS}")
 
-add_subdirectory(third_party)
 add_subdirectory(torchtext/csrc)
diff --git a/torchtext/csrc/CMakeLists.txt b/torchtext/csrc/CMakeLists.txt
index 5d67bba8adc4..1aa13dab7f42 100644
--- a/torchtext/csrc/CMakeLists.txt
+++ b/torchtext/csrc/CMakeLists.txt
@@ -18,10 +18,6 @@ set(
 set(
   LIBTORCHTEXT_INCLUDE_DIRS
   ${PROJECT_SOURCE_DIR}
-  ${PROJECT_SOURCE_DIR}/third_party/sentencepiece/src
-  $<TARGET_PROPERTY:re2,INCLUDE_DIRECTORIES>
-  $<TARGET_PROPERTY:double-conversion,INCLUDE_DIRECTORIES>
-  $<TARGET_PROPERTY:utf8proc,INCLUDE_DIRECTORIES>
   ${TORCH_INSTALL_PREFIX}/include
   ${TORCH_INSTALL_PREFIX}/include/torch/csrc/api/include
   )
@@ -47,7 +43,7 @@ function (define_library name source include_dirs link_libraries compile_defs)
   target_include_directories(${name} PRIVATE ${include_dirs})
   target_link_libraries(${name} ${link_libraries})
   target_compile_definitions(${name} PRIVATE ${compile_defs})
-  set_target_properties(${name} PROPERTIES PREFIX "")
+  set_target_properties(${name} PROPERTIES PREFIX "" SOVERSION 0.16)
   if (MSVC)
     set_target_properties(${name} PROPERTIES SUFFIX ".pyd")
   endif(MSVC)
@@ -119,10 +115,6 @@ if (BUILD_TORCHTEXT_PYTHON_EXTENSION)
   set(
     EXTENSION_INCLUDE_DIRS
     ${PROJECT_SOURCE_DIR}
-    ${PROJECT_SOURCE_DIR}/third_party/sentencepiece/src
-    $<TARGET_PROPERTY:re2,INCLUDE_DIRECTORIES>
-    $<TARGET_PROPERTY:double-conversion,INCLUDE_DIRECTORIES>
-    $<TARGET_PROPERTY:utf8proc,INCLUDE_DIRECTORIES>
     ${TORCH_INSTALL_PREFIX}/include
     ${TORCH_INSTALL_PREFIX}/include/torch/csrc/api/include
     )
-- 
2.44.0

